### ISSUE TITLE
Is the Performance of "Unfrozen w & Unfrozen q" Actually Inferior in the Final Comparative Experiments?

### ISSUE THREAD

**Issue by hertz-pj** [youngsheen/SimVQ#1]
![企业微信截图_17308187715975](https://github.com/user-attachments/assets/c65d8e9d-ae8a-43ea-90ba-bf591347ec27)
Hi,

From the experimental figure provided, it seems that the effect of unfrozen w & unfrozen q does not appear to be poor, as the loss is sufficiently low and the norm w values also do not reflect the final actual performance. However, I was wondering if the actual inferior performance is demonstrated in the final comparative experiments through FID for images and PSEQ for speech quality?

Could you please provide clarification or additional data reflecting the actual performance differences?

Thank you!

**Comment by youngsheen** [comment]
Hi,

In the above figure of toy examples, the loss curves for the unfrozen  w  and unfrozen  q  rapidly approach zero, while the linear matrix  w  shows almost no significant change, which means q **dominates** the optimization dynamics. This clearly indicates the problem of representation collapse, as two points quickly reach the target while almost all other points remain stationary (or not "utilized", see Fig3a). Note that the meaning of "loss" here is different from the reconstruction loss, but similar to the codebook commitment loss. In this toy example, the loss indicates that, although the optimization process seems to reach the optimum quickly, a representation collapse has actually occurred. This is not the representation learning result we aim for. Ideally, we want all points to converge toward the target points and be "utilized" effectively (Fig2b).

In Table 3 of the paper, we compare the two states of  q  frozen and  q  unfrozen. It can be seen that the final rFID is better when  q  is frozen.  Moreover, due to q is frozen and not updated by gradient , it also offers higher computational efficiency.

**Comment by hertz-pj** [comment]
Thank you for your response. Could you please clarify how the e_dim hyperparameter is set when training the speech SimVQ model?

**Comment by youngsheen** [comment]
We did not reduce the code dimension, so e_dim equals encoder dim, you can find it here https://github.com/youngsheen/SimVQ/blob/main/configs/libritts_24khz.yaml#L38
